pipelines/folioRancher/folioNamespaceTools/manageNamespace/createDailySnapshotEureka/Jenkinsfile

```diff
@@ -0,0 +1,112 @@
+                        booleanParam(name: 'RW_SPLIT', defaultValue: false, description: '(Optional) Set to true to enable Read/Write split'),
+                        folioParameters.pgType(),
+                        folioParameters.kafkaType(),
+                        folioParameters.opensearchType(['aws']),
+                        folioParameters.s3Type(),
+                        folioParameters.agent(),
+                        folioParameters.refreshParameters()]),
+            pipelineTriggers([parameterizedCron('''45 21 * * * %CLUSTER=folio-etesting;NAMESPACE=snapshot;EUREKA=true;CONFIG_TYPE=testing;CONSORTIA=false;RW_SPLIT=false;POSTGRESQL=built-in;KAFKA=built-in;OPENSEARCH=aws;S3_BUCKET=built-in;AGENT=rancher''')])])
+
+
+if (params.REFRESH_PARAMETERS) {
+  currentBuild.result = 'ABORTED'
+  return
+}
+
+final String deleteNamespaceJobName = '/folioRancher/tmpFolderForDraftPipelines/Eureka/deleteNamespace-Eureka'
+
+CreateNamespaceParameters namespaceBaseParams = new CreateNamespaceParameters.Builder()
+  .clusterName('folio-etesting')
+  .namespaceName('snapshot')
+  .configType('testing')
+  .rwSplit(false)
+  .greenmail(false)
+  .mockServer(false)
+  .pgType('built-in')
+  .pgVersion('13.13')
+  .kafkaType('built-in')
+  .opensearchType('aws')
+  .s3Type('built-in')
+  .members('')
+  .worker('rancher')
+  .build()
+
+
+CreateNamespaceParameters namespaceParams = new CreateNamespaceParameters.Builder()
+  .clusterName('folio-etesting')
+  .namespaceName('snapshot')
+  .folioBranch('snapshot')
+  .okapiVersion('latest')
+  .configType('testing')
+  .loadReference(true)
+  .loadSample(true)
+  .consortia(true)
+  .eureka(true)
+  .rwSplit(false)
+  .greenmail(false)
+  .mockServer(false)
+  .rtr(false)
+  .pgType('built-in')
+  .pgVersion('13.13')
+  .kafkaType('built-in')
+  .opensearchType('aws')
+  .s3Type('built-in')
+  .members('thunderjet,folijet,spitfire,vega,thor,Eureka,volaris,corsair,bama,aggies,dreamliner,leipzig')
+  .uiBuild(true)
+  .build()
+
+ansiColor('xterm') {
+  node(params.AGENT) {
+    try {
+      stage('Ini') {
+        buildName "${namespaceBaseParams.getClusterName()}-${namespaceBaseParams.getNamespaceName()}.${env.BUILD_ID}"
+        buildDescription "Config: ${namespaceBaseParams.getConfigType()}"
+      }
+
+      stage('Checkout') {
+        checkout scm
+      }
+
+      stage('[TF] Destroy') {
+        def nsExists
+        folioHelm.withKubeConfig(namespaceBaseParams.getClusterName()) {
+          nsExists = kubectl.checkNamespaceExistence("${namespaceBaseParams.getNamespaceName()}")
+        }
+        if (nsExists) {
+          folioTriggerJob.deleteNamespace(deleteNamespaceJobName, namespaceBaseParams)
+        } else {
+          new Logger(this, 'dailySnapshotEureka').warning("${namespaceBaseParams.getNamespaceName()} namespace does not exists!")
+        }
+      }
+
+      if (namespaceParams.eureka) {
```

Comment:
But why we create a new Namespace outside of Stage?

Reply:
Because it's a separate part of a code residing in folioNamespaceCreateEureka or folioNamespaceCreate var, it has its stages inside. This is a common approach in our pipelines.